cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -117,7 +141,8 @@ public void setConnectionConfig(@Nullable ConnectionConfig connectionConfig) {
   public void tryConnect(ConnectionConfig connectionConfig, PrintStream output, boolean debug) throws Exception {
     try {
       AccessToken accessToken = acquireAccessToken(clientConfig, connectionConfig, output, debug);
-      checkConnection(clientConfig, connectionConfig, accessToken);
+      connectionConfig = clientConfig.getConnectionConfig();
```

Comment:
I think the old "checkConnection" and ClientConfig is not good. If we use checkConnection to validate a connection before it can be set into ClientConfig, then checkConnection should take an incomplete "ClientConfig" (ideally the ClientConfig should be an immutable object). Refactoring is needed on both of them.